core-api/src/test/java/com/optimizely/ab/config/ExperimentTest.java

```diff
@@ -0,0 +1,195 @@
+        scenario3List.add(one);
+        scenario3List.add(two);
+        scenario3List.add(three);
+        AndCondition scenario3 = new AndCondition(scenario3List);
+
+        // Scenario 4 - ["not", "1"]
+        NotCondition scenario4 = new NotCondition(one);
+
+        // Scenario 5 - ["or", "1"]
+        List<Condition> scenario5List = new ArrayList<>();
+        scenario5List.add(one);
+        OrCondition scenario5 = new OrCondition(scenario5List);
+
+        // Scenario 6 - ["and", "1"]
+        List<Condition> scenario6List = new ArrayList<>();
+        scenario6List.add(one);
+        AndCondition scenario6 = new AndCondition(scenario6List);
+
+        // Scenario 7 - ["1"]
+        AudienceIdCondition scenario7 = one;
+
+        // Scenario 8 - ["1", "2"]
+        // Defaults to Or in Datafile Parsing resulting in an OrCondition
+        // Same as Scenario 2
+
+        OrCondition scenario8 = scenario2;
+
+        // Scenario 9 - ["and", ["or", "1", "2"], "3"]
+        List<Condition> Scenario9List = new ArrayList<>();
+        Scenario9List.add(scenario2);
+        Scenario9List.add(three);
+        AndCondition scenario9 = new AndCondition(Scenario9List);
+
+        // Scenario 10 - ["and", ["or", "1", ["and", "2", "3"]], ["and", "11, ["or", "12", "13"]]]
+        List<Condition> scenario10List = new ArrayList<>();
+
+        List<Condition> or1213List = new ArrayList<>();
+        or1213List.add(twelve);
+        or1213List.add(thirteen);
+        OrCondition or1213 = new OrCondition(or1213List);
+
+        List<Condition> and11Or1213List = new ArrayList<>();
+        and11Or1213List.add(eleven);
+        and11Or1213List.add(or1213);
+        AndCondition and11Or1213 = new AndCondition(and11Or1213List);
+
+        List<Condition> and23List = new ArrayList<>();
+        and23List.add(two);
+        and23List.add(three);
+        AndCondition and23 = new AndCondition(and23List);
+
+        List<Condition> or1And23List = new ArrayList<>();
+        or1And23List.add(one);
+        or1And23List.add(and23);
+        OrCondition or1And23 = new OrCondition(or1And23List);
+
+        scenario10List.add(or1And23);
+        scenario10List.add(and11Or1213);
+        AndCondition scenario10 = new AndCondition(scenario10List);
+
+        // Scenario 11 - ["not", ["and", "1", "2"]]
+        List<Condition> and12List = new ArrayList<>();
+        and12List.add(one);
+        and12List.add(two);
+        AndCondition and12 = new AndCondition(and12List);
+
+        NotCondition scenario11 = new NotCondition(and12);
+
+        // Scenario 12 - ["or", "1", "100000"]
+        List<Condition> scenario12List = new ArrayList<>();
+        scenario12List.add(one);
+        AudienceIdCondition unknownAudience = new AudienceIdCondition("100000");
+        scenario12List.add(unknownAudience);
+
+        OrCondition scenario12 = new OrCondition(scenario12List);
+
+        // Scenario 13 - Empty String "" already accounted for in Datafile parsing
+
```

Comment:
Let's add an invalid audienceCondition for 13: ["and", "and"] and validate it returns an empty string.

Reply:
Added scenario13 as described